src/main/java/cc/polyfrost/oneconfig/config/Config.java

```diff
@@ -273,6 +287,131 @@ protected final void generateOptionList(Object instance, Class<?> targetClass, O
+        logger.trace("Generating option list for {}... subcategory (targetting={})", mod.name, targetClass.getName());
+
+        String pagePath = page.equals(mod.defaultPage) ? "" : page.name + ".";
+        for (Field field : targetClass.getDeclaredFields()) {
+            Option option = ConfigUtils.findAnnotation(field, Option.class);
+            CustomOption customOption = ConfigUtils.findAnnotation(field, CustomOption.class);
+            String optionName = pagePath + field.getName();
+            if (option != null) {
+                BasicOption configOption = ConfigUtils.addOptionToPage(page, option, field, instance, category, subcategory, migrate ? mod.migrator : null);
+                optionNames.put(optionName, configOption);
+            } else if (customOption != null) {
+                BasicOption configOption = getCustomOption(field, customOption, page, mod, migrate);
+                if (configOption == null) continue;
+                optionNames.put(optionName, configOption);
+            } else if (field.isAnnotationPresent(Page.class)) {
+                Page optionPage = field.getAnnotation(Page.class);
+                OptionSubcategory pageSubcategory = ConfigUtils.getSubCategory(page, category, subcategory);
+                Object pageInstance = ConfigUtils.getField(field, instance);
+                if (pageInstance == null) continue;
+                ConfigPageButton button;
+                if (pageInstance instanceof cc.polyfrost.oneconfig.gui.pages.Page) {
+                    button = new ConfigPageButton(field, instance, optionPage.name(), optionPage.description(), category, subcategory, (cc.polyfrost.oneconfig.gui.pages.Page) pageInstance);
+                } else {
+                    OptionPage newPage = new OptionPage(optionPage.name(), mod);
+                    generateOptionList(pageInstance, newPage, mod, migrate);
+                    button = new ConfigPageButton(field, instance, optionPage.name(), optionPage.description(), category, subcategory, newPage);
+                }
+                if (optionPage.location() == PageLocation.TOP) pageSubcategory.topButtons.add(button);
+                else pageSubcategory.bottomButtons.add(button);
+            } else if (field.isAnnotationPresent(HUD.class)) {
+                HUDUtils.addHudOptions(page, field, instance, this);
+            }
+        }
+        for (Method method : targetClass.getDeclaredMethods()) {
+            Button button = ConfigUtils.findAnnotation(method, Button.class);
+            String optionName = pagePath + method.getName();
+            if (button != null) {
+                BasicOption option = ConfigUtils.addOptionToPage(page, method, instance, category, subcategory);
+                optionNames.put(optionName, option);
+            }
+        }
+        logger.trace("Finished generating option list for {} subcategory (targetting={})", mod.name, targetClass.getName());
```

Comment:
a lot of this code is reused from `addCategory`, maybe they could be combined

Reply:
they look simular but they pass a diffrent amount of arguments to some methods, so it wouldnt really work. i could try tho

Reply:
i presume its not possible to abstractify this? they really do look very similar